pkg/instance/service.go

```diff
@@ -773,3 +774,67 @@ func (s Service) Reset(ctx context.Context, token string, instance *model.Deploy
 
 	return s.deployDeploymentInstance(ctx, token, instance, ttl)
 }
+
+func (s Service) FilestoreBackup(ctx context.Context, instance *model.DeploymentInstance, name string, database *model.Database) error {
+	s.logger.InfoContext(ctx, "save again", "database", database)
+
+	endpoint := fmt.Sprintf("%s-minio.%s.svc:9000", instance.Name, instance.GroupName)
+	minioClient, err := newMinioClient("dhisdhis", "dhisdhis", endpoint, false)
+	if err != nil {
+		return err
+	}
+
+	backupService := NewBackupService(s.logger, minioClient, s.s3Client)
+
+	name = strings.TrimSuffix(name, ".pgc")
+	name = strings.TrimSuffix(name, ".tar.gz")
+	key := fmt.Sprintf("%s/%s-%s.tar.gz", instance.GroupName, name, "fs")
+	err = backupService.PerformBackup(ctx, "dhis2", s.s3Bucket, key)
+	if err != nil {
+		return err
+	}
+
+	// Record backup in database
+	s3Uri := fmt.Sprintf("s3://%s/%s", s.s3Bucket, key)
+	filestore, err := s.recordBackup(ctx, instance.GroupName, s3Uri, name+"-fs.tar.gz")
+	if err != nil {
+		return err
+	}
+
+	database.FilestoreID = filestore.ID
+
+	s.logger.InfoContext(ctx, "save2", "database.FilestoreID", database.FilestoreID)
+	s.logger.InfoContext(ctx, "save3", "filestore.ID", filestore.ID)
```

Comment:
After testing trying out the saving functionality I noticed these lines in the logs and they don't make much sense.
Here's how they  looks:
```
{"time":"2025-01-23T17:51:57.078853857Z","level":"INFO","msg":"save2","database.FilestoreID":8,"correlationId":"084f89d2-97cd-4924-9130-19cf508a12b3","user":1}
{"time":"2025-01-23T17:51:57.078869032Z","level":"INFO","msg":"save3","filestore.ID":8,"correlationId":"084f89d2-97cd-4924-9130-19cf508a12b3","user":1}
```